internal/pkg/action/operator_uninstall.go

```diff
@@ -71,12 +70,8 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 	// deleting the subscription.
 	var crds, csvs, others []controllerutil.Object
 	if sub.Status.InstallPlanRef != nil {
```

Comment:
I don't think we need to pay attention to `sub.Status.InstallPlanRef` anymore, since we don't use it to figure out the install plan key.

My instinct here is to locate the install plan separately from getting its resources, and change this `if` statement to check whether we found an install plan.

So something like this:
```go
ip, err := u.getInstallPlanForCSV(sub.Status.StartingCSV)
if err != nil { ... }

var crds, csvs, others []controllerutil.Object
if ip != nil {
        var err error
        crds, csvs, others = u.getInstallPlanResources(*ip)  
}
```